tensorflow-ops/src/TensorFlow/Ops.hs

```diff
@@ -256,6 +257,15 @@ constant (Shape shape') values
                       [def & TensorShape.size .~ x | x <- shape']
                 & tensorVal .~ values
 
+-- | Reshape a N-D tensor down to a scalar.
+toScalar :: (TensorType a) 
+         => Tensor v a 
+         -> Tensor Value a
+toScalar = flip CoreOps.reshape (vector scalarShape)
```

Comment:
I'd avoid flip here. 
toScalar x = CoreOps.reshape x (vector scalarShape)
Is both shorter and easier to understand.